internal/controller/pdbwatcher_controller.go

```diff
@@ -143,28 +143,12 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if err != nil {
 			logger.Error(err, "failed to update deployment")
 			return ctrl.Result{}, err
-		}
-
-		// Save ResourceVersion to PDBWatcher status this will cause another reconcile.
-		pdbWatcher.Status.TargetGeneration = target.Obj().GetGeneration()
-		pdbWatcher.Status.LastEviction = pdbWatcher.Spec.LastEviction //we could still keep a log here if thats useful
-		//should we clear evictions?
-		err = r.Status().Update(ctx, pdbWatcher)
-		if err != nil {
-			logger.Error(err, "Failed to update PDBWatcher status")
-			return ctrl.Result{}, err
-		}
-
+		}	
+		
 		// Log the scaling action
 		logger.Info(fmt.Sprintf("Scaled up Deployment %s/%s to %d replicas", target.Obj().GetNamespace(), target.Obj().GetName(), newReplicas))
-		return ctrl.Result{}, nil
-
-	}
-
-	// Watch for changes in PDB to revert to original state
-	if target.GetReplicas() != pdbWatcher.Status.MinReplicas {
-
-		// Revert Target to the original state
+	} else  if target.GetReplicas() != pdbWatcher.Status.MinReplicas {
+		//okay we aren't at allowed disruptions Revert Target to the original state
 		// This is bad if its a statefulset because scale down removes non zero replicas first even if zero isn't ready
```

Comment:
this is true for deployments as well. We should wait to scale down till after the eviction has had a chance to go through completlely.

Reply:
Fixed this.